server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java

```diff
@@ -682,5 +641,53 @@ long getInitialTermBeforeJoiningALeader() {
         public void close() {
             assertTrue(openPersistedStates.remove(this));
         }
+
+        @Override
+        public void getLatestStoredState(ActionListener<ClusterState> listener) {
+            ActionListener.completeWith(listener, () -> {
+                var latestClusterState = sharedStore.getLatestClusterState();
+                if (latestClusterState == null) {
+                    return null;
+                }
+
+                if (isLatestAcceptedStateStale(latestClusterState) == false) {
+                    return null;
+                }
+
+                if (latestClusterState.term() > currentTerm) {
+                    return null;
+                }
+
+                return ClusterStateUpdaters.recoverClusterBlocks(
+                    ClusterStateUpdaters.addStateNotRecoveredBlock(
+                        ClusterState.builder(new ClusterName("elasticsearch"))
```

Comment:
Really we should get this from settings, but we don't use any nonstandard cluster names in these tests so I think this is ok too:

```suggestion
                        ClusterState.builder(ClusterName.DEFAULT)
```